azkaban-common/src/main/java/azkaban/utils/AzkabanPatternLayout.java

```diff
@@ -0,0 +1,112 @@
+package azkaban.utils;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import org.apache.log4j.helpers.PatternConverter;
+import org.apache.log4j.helpers.PatternParser;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+
+/**
+ * When we use the log4j Kafka appender, it seems that the appender simply does not log the stack trace anywhere
+ * Seeing as the stack trace is a very important piece of information, we create our own PatternLayout class that
+ * appends the stack trace to the log message that reported it, so that all the information regarding that error
+ * can be found one in place. In addition, we create a new PatternParser to deliver logs with current host name.
+ */
+public class AzkabanPatternLayout extends PatternLayout {
```

Comment:
I was thinking this class could be supplemented with other custom patterns, and then could be used inside azkaban-server.log as well.